test/connectObservable.test.tsx

```diff
@@ -136,47 +198,52 @@ describe("connectObservable", () => {
 
     render(<TestSuspense />)
 
-    expect(screen.queryByText("Result")).toBeNull()
-    expect(screen.queryByText("Waiting")).not.toBeNull()
+    expect(screen.queryByText("Result 0")).not.toBeNull()
+    expect(screen.queryByText("Waiting")).toBeNull()
 
     fireEvent.click(screen.getByText(/Next/i))
 
-    expect(screen.queryByText("Result")).toBeNull()
     expect(screen.queryByText("Waiting")).not.toBeNull()
 
     await wait(110)
 
     expect(screen.queryByText("Result 1")).not.toBeNull()
     expect(screen.queryByText("Waiting")).toBeNull()
+  })
 
-    fireEvent.click(screen.getByText(/Next/i))
-
-    expect(screen.queryByText("Result")).toBeNull()
-    expect(screen.queryByText("Waiting")).not.toBeNull()
+  it("keeps in suspense if more than two SUSPENSE are emitted in succesion", async () => {
+    const source$ = concat(
+      of(0),
+      of(SUSPENSE).pipe(delay(50)),
+      of(SUSPENSE).pipe(delay(50)),
+      of(1).pipe(delay(50)),
+    )
+    const [useDelayedNumber] = connectObservable(source$)
+    const Result: React.FC = () => <div>Result {useDelayedNumber()}</div>
+    const TestSuspense: React.FC = () => {
+      return (
+        <Suspense fallback={<span>Waiting</span>}>
+          <Result />
+        </Suspense>
+      )
+    }
 
-    await wait(110)
+    render(<TestSuspense />)
 
-    expect(screen.queryByText("Result 2")).not.toBeNull()
+    expect(screen.queryByText("Result 0")).not.toBeNull()
     expect(screen.queryByText("Waiting")).toBeNull()
 
-    fireEvent.click(screen.getByText(/Next/i))
+    await wait(60)
```

Comment:
Let's use subjects rather than `wait`. `wait` ends up generating flaky tests and slowing tests down.